### ISSUE TITLE
Incorporate support for Strings - Unicode

### ISSUE THREAD

**Issue by pjljvandelaar** [TorXakis/TorXakis#418]
Can the basictype string be a list of unicode characters (instead of ascii)?
Does Haskell support unicode?
Does smt-lib support unicode?
Does a SMT problem-solver exist that support unicode?

**Comment by pjljvandelaar** [comment]
Which basic datatypes should torxakis support? See also #287 and #410

**Comment by capitanbatata** [comment]
>  Does Haskell support unicode?

Yes, the [`text`](https://hackage.haskell.org/package/text) package does.

I'm labeling this as a (potential) performance improvement item, since we will move from `String` to `Text`.

**Comment by pjljvandelaar** [comment]
We also need to consider the support of CVC4 and Z3 for Unicode.

See e.g. limitations on http://cvc4.cs.stanford.edu/wiki/Strings:
`The current version of the solver supports ASCII characters only. We might move on to UNICODE in future versions. `

and https://rise4fun.com/z3/tutorialcontent/sequences who only talks about ASCII

**Comment by pjljvandelaar** [comment]
See    http://smtlib.cs.uiowa.edu/theories-UnicodeStrings.shtml for the specification of an SMT-LIB 2 theory of strings and regular expressions.
